src/lib/version.ts

```diff
@@ -0,0 +1,132 @@
+import debugFactory from 'debug';
+import semver from 'semver';
+
+const debug = debugFactory('clark:lib:version');
+
+/**
+ * Selcts the greater of two semver ranges combined with their most permissive
+ * range modifier
+ * @param left
+ * @param right
+ */
+export function select(left: string | null, right: string | null): string {
```

Comment:
```tsx
export function select(left: string | undefined, right: string | undefined): string {
  debug(`checking if "${left}" and "${right}" are compatible`);

  if (!left && !right) {
    throw new Error('Cannot select a version from "null" and "null"');
  }
  
  return right || left!;
}
```

If you know better than the type system, you can use the [Non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator).

To do this without casting:

```
  const version = left || right;

  if (!version) {
    throw new Error('Cannot select a version from "null" and "null"');
  }
  
  return version;
```